src/main/java/com/minecolonies/coremod/colony/buildings/moduleviews/MinerAssignmentModuleView.java

```diff
@@ -0,0 +1,142 @@
+package com.minecolonies.coremod.colony.buildings.moduleviews;
+
+import com.ldtteam.blockout.views.Window;
+import com.minecolonies.api.colony.ICitizenDataView;
+import com.minecolonies.api.colony.buildings.HiringMode;
+import com.minecolonies.api.colony.buildings.modules.AbstractBuildingModuleView;
+import com.minecolonies.api.colony.buildings.modules.IAssignmentModuleView;
+import com.minecolonies.api.colony.buildings.views.IBuildingView;
+import com.minecolonies.api.colony.jobs.ModJobs;
+import com.minecolonies.api.colony.jobs.registry.JobEntry;
+import com.minecolonies.api.util.constant.Constants;
+import com.minecolonies.coremod.Network;
+import com.minecolonies.coremod.client.gui.modules.SpecialAssignmentModuleWindow;
+import com.minecolonies.coremod.network.messages.server.colony.building.HireFireMessage;
+import com.minecolonies.coremod.network.messages.server.colony.building.QuarryHiringModeMessage;
+import net.minecraft.network.PacketBuffer;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Module view for miner assignment.
+ */
+public class MinerAssignmentModuleView extends AbstractBuildingModuleView implements IAssignmentModuleView
+{
+    /**
+     * List of the worker ids.
+     */
+    private final Set<Integer> workerIDs = new HashSet<>();
+
+    /**
+     * The hiring mode of the building.
+     */
+    private HiringMode hiringMode;
+
+    @Override
+    public List<Integer> getAssignedCitizens()
+    {
+        return new ArrayList<>(workerIDs);
+    }
+
+    @Override
+    public void addCitizen(final @NotNull ICitizenDataView citizen)
+    {
+        workerIDs.add(citizen.getId());
+        Network.getNetwork().sendToServer(new HireFireMessage(buildingView, true, citizen.getId(), getJobEntry()));
+    }
+
+    @Override
+    public void deserialize(@NotNull final PacketBuffer buf)
+    {
+        final int size = buf.readInt();
```

Comment:
no super.deserialize intended?

Reply:
There is no super class with deserialize here